frontend/workflows/projectSelector/src/project-group.tsx

```diff
@@ -153,53 +154,51 @@ const ProjectGroup: React.FC<ProjectGroupProps> = ({ title, group, displayToggle
           {numProjects === 0 && (
             <StyledNoProjectsText>No projects in this group yet.</StyledNoProjectsText>
           )}
-          {Object.keys(state[group])
-            .sort()
-            .map(key => (
-              <StyledMenuItem key={key}>
-                <Checkbox
-                  name={key}
-                  size="small"
-                  disabled={state.loading}
-                  onChange={() =>
+          {groupKeys.sort().map(key => (
+            <StyledMenuItem key={key}>
+              <Checkbox
+                name={key}
+                size="small"
+                disabled={state?.loading}
```

Comment:
i think you can eliminate the added `state?` checks by throwing an error from the hook and never returning an undefined state, which should never happen

Reply:
If we don't expect state to ever be undefined should we change the initialization of it (and types) to not allow for undefined?

Reply:
@dschaller , just for my understanding, do you mean something like this?
```
const StateContext = React.createContext({
  [Group.PROJECTS]: {},
  [Group.UPSTREAM]: {},
  [Group.DOWNSTREAM]: {},
  projectData: {},
  loading: false,
  error: undefined,
} as State);
```
we also initialize it with default values in the project selector component in `useReducer`

Reply:
Chatted IRL with @scarlettperry and we are going to punt on removing undefined as a type until later.